### ISSUE TITLE
Allow blacklisted names in tests

### ISSUE THREAD

**Issue by mikeyhew** [rust-lang/rust-clippy#2198]
Hi! First of all, great job with clippy to everyone involved. I just installed it and ran `cargo clippy`, and I like how seamlessly it runs – it's just like running `cargo check`, but with more lints.

One of the lints that I noticed is the lint for placeholder variable names like `foo` and `bar`. In general I think this is a great idea, but in the context of tests I think it should be turned off by default. I use `foo` and `bar` all the time in tests, and I don't think it's a problem there. Does anyone think otherwise?

**Comment by oli-obk** [comment]
Makes sense to me. Not sure how to solve it properly though. We can disable the lints via flags when `cargo clippy` is linting tests. For unit tests we'd need to add some way to prevent linting inside anything that has the `#[test]` attribute

**Comment by ada4a** [comment]
I think this can be closed thanks to #7379
